model/src/main/kotlin/utils/PurlUtils.kt

```diff
@@ -91,7 +101,7 @@ fun Identifier.getPurlType() =
  * [in the documentation](https://github.com/package-url/purl-spec/blob/master/README.rst#purl).
  * E.g. 'maven' for Gradle projects.
  */
-fun Identifier.toPurl() = if (this == Identifier.EMPTY) "" else createPurl(getPurlType(), namespace, name, version)
+fun Identifier.toPurl() = toPurl(emptyMap(), "")
```

Comment:
Would it make sense to, instead of an overload, just add these two parameter to this function, with default values `emptyMap` and `""` ?

Reply:
That's how it was before [the force-push](https://github.com/oss-review-toolkit/ort/compare/ec7acb0377d80507a25bb5fbc199ad96e7336948..0f484d684dfc91c1c951c7ea27411680040cead1), and I also liked that better... @nnobelis why did you change it?

Reply:
As explained in the commit message it broke the FreeMarker templates that call `toUrl` (the tests failed): I guess Kotlin's implementation of extension functions with default values is not compatible with FreeMarker Java's reflection.

Reply:
Have you considered annotation the function (from before) with `@JvmOverloads`? I might solve the problem as well.

Reply:
I think we can go around with `@JvmOverloads`. Would it be ok to put this annotation on `toUrl` ?

Reply:
Where does `toUrl` now come from? Did you maybe mean this?

```
@JvmOverloads
fun Identifier.toPurl(qualifiers: Map<String, String> = emptyMap(), subpath: String = "") = ...
```

Reply:
Yes `toPurl`. Sorry typo.

Reply:
If above signature works, IMO it'd bee good.

Reply:
@nnobelis, please also explain in the commit message what `@JvmOverloads` is needed for.